### PR TITLE
[W4.4b][T15-A1] Chu Qinghao

### PR DESCRIPTION
**Pull request by qinghao1**
[LO-ImplementClass]

### CODE REVIEW

**Comment by qinghao1** [comment]
Wrote my git config wrongly so the commits show up as a different person

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -2,15 +2,33 @@
...
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in the format of: " +
+            "'a/BLOCK, STREET, UNIT, POSTAL_CODE'";
+    public static final String ADDRESS_VALIDATION_REGEX = "(.+),(.+),(.+),(.+)";
+
+    private static final String ADDRESS_SPLIT_DELIMITER = ",";
+
+    private static final Integer BLOCK_INDEX = 0;
```
Comment (madsonic) [inline src/seedu/addressbook/data/person/Address.java:23]: Would `int` be better here?

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -27,6 +45,14 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        String[] splitAddress = trimmedAddress.split(ADDRESS_SPLIT_DELIMITER);
+        if (splitAddress.length < 4) { //Sanity check
```
Comment (madsonic) [inline src/seedu/addressbook/data/person/Address.java:49]: Sanity check is good practice! Would you want to consider length of more than 4 legal though?

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -38,7 +64,11 @@ public static boolean isValidAddress(String test) {
...
     public String toString() {
-        return value;
+        return String.format("%s, %s, %s, %s",
+                block.toString(),
+                street.toString(),
+                unit.toString(),
+                postalCode.toString());
     }
 
     @Override
```
Comment (madsonic) [inline src/seedu/addressbook/data/person/Address.java:74]: Now that you decomposed address into individual components, is there a need to update this method? 
Can you think of 2 addresses that have equal individual components yet this method return false?

**Review by madsonic — COMMENTED** [review]
Great use of composition! Remember to add tests for your new components. They will help you uncover elusive bugs. Keep up the good work!

**Comment by madsonic** [comment]
Close this PR if you have no other questions

**Pull request closed without merging**